bob/src/hw_metrics_collector.rs

```diff
@@ -235,54 +323,94 @@ impl DiskStatCollector {
+            if new_ds.extended {
+                if let (Ok(w_ticks), Ok(r_ticks)) = (
+                    // time spend on write is in 10th column
+                    parts[10].parse::<u128>(),
+                    // time spend on read is in 6th column
+                    parts[6].parse::<u128>(),
+                ) {
+                    new_ds.read_time = r_ticks;
+                    new_ds.write_time = w_ticks;
+                } else {
+                    let msg = format!("Can't parse {} values to unsigned int", DISK_STAT_FILE);
+                    return Err(CommandError::Primary(msg));
+                }
+            }
+        } else {
+            let msg = format!("Can't parse {} values to unsigned int", DISK_STAT_FILE);
+            return Err(CommandError::Primary(msg));
+        }
+        Ok(new_ds)
+    }
 
-                    if let (Ok(rwait), Ok(wwait)) = (
-                        // readops wait time is in 5th column
-                        Self::parse_iostat_result(lsp[5]),
-                        // writeops wait time is in 11th column
-                        Self::parse_iostat_result(lsp[11]),
-                    ) {
-                        let iowait = rwait + wwait;
-
-                        let gauge_name = format!("{}_iowait", metric_prefix);
-                        gauge!(gauge_name, iowait);
-                    } else {
-                        return Err(CommandError::Primary("Can't parse iostat output into float".to_string()));
+    fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
+        let diskstats = self.diskstats()?;
+        let now = Instant::now();
+        let elapsed = now.duration_since(self.upd_timestamp);
+        if elapsed > DIFFCONTAINER_THRESHOLD {
+            self.upd_timestamp = now;
+            let elapsed = elapsed.as_secs_f64();
+            for i in diskstats {
+                let lsp: Vec<&str> = i.split_whitespace().collect();
+                if lsp.len() >= 8 {
+                    // compare device name from 2nd column with disk device names
+                    if let Some(ds) = self.disk_metric_data.get_mut(lsp[2]) {
+                        let new_ds = Self::parse_stat_line(lsp)?;
+                        
+                        if let Some(diff) = ds.stats.diff(new_ds) {
+                            let iops = (diff.reads + diff.writes) as f64 / elapsed;
+                            let gauge_name = format!("{}_iops", ds.prefix);
+                            gauge!(gauge_name, iops);
+
+                            if diff.extended {
+                                let mut iowait = 0.;
+                                if diff.reads > 0 {
+                                    iowait += diff.read_time as f64 / diff.reads as f64;
+                                }
+                                if diff.writes > 0 {
+                                    iowait += diff.write_time as f64 / diff.writes as f64;
```

Comment:
I just notice, that this calculation of `iowait` is incorrect. `iowait` is the percentage of CPU time spent waiting for an I/O operation to complete. What you calculate is just an average time of single I/O operation. 

I suggest to calculate disk utilization instead (`util`). It is the time spent doing I/Os (column 13) divided by the elapsed time in milliseconds